backend/corpora/common/dataset_validator.py

```diff
@@ -167,8 +184,75 @@ def verify_uns(self, data_object: anndata.AnnData):
             CorporaConstants.REQUIRED_DATASET_METADATA_FIELDS
             + CorporaConstants.REQUIRED_DATASET_PRESENTATION_METADATA_FIELDS
         ):
-            if metadata_field not in unstructured_metadata_keys:
-                self.log_error_message(metadata_field, "uns", type(data_object).__name__)
+            if metadata_field.field_name not in unstructured_metadata_keys:
+                self.log_error_message(metadata_field.field_name, "uns", type(data_object).__name__)
+            else:
+                self.verify_metadata_type(metadata_field, data_object.uns.get(metadata_field.field_name))
+
+    def verify_metadata_type(self, metadata_property, metadata_values_in_dataset):
```

Comment:
Could you add typing hints to function signatures? Personally find it helps with readability, thanks!

Reply:
Ooh, just got to these later in the PR, and assuming they can be "typed" given the `isinstance` usage, but it would be cool for example:

`..., metadata_property: typing.Union[Ontology, TypedMetadata],`

Reply:
Done!